src/ui/React/LinearForwardProgress.tsx

```diff
@@ -0,0 +1,75 @@
+import * as React from "react";
+import { useRef } from "react";
+import LinearProgress from "@mui/material/LinearProgress";
+
+interface LinearForwardProgressProps {
+  value: number;
+  backgroundColor?: React.CSSProperties["color"];
+  barColor?: React.CSSProperties["color"];
+}
+
+export const LinearForwardProgress = React.forwardRef<unknown, LinearForwardProgressProps>(
+  function LinearForwardProgress({ value, backgroundColor, barColor, ...otherProps }, ref): React.ReactElement {
+    const previousValueRef = useRef(value);
+    const animationKeyRef = useRef(0);
+
+    const isDecreasing = value < previousValueRef.current;
+    const previousValue = previousValueRef.current;
+
+    // Update ref for next render
+    previousValueRef.current = value;
+
+    if (isDecreasing) {
+      // Increment animation key to force new CSS animation
+      animationKeyRef.current += 1;
+
+      // MUI LinearProgress uses translateX where:
+      // translateX(-100%) = 0% progress, translateX(0%) = 100% progress
+      const startPos = previousValue - 100;
+      const endPos = value - 100;
+      const animationName = `forwardAnim${animationKeyRef.current}`;
+
+      // Calculate timing: proportional split between fill-to-100% and reset-to-final phases
+      const firstPhasePercent = ((100 - previousValue) / (100 - previousValue + value)).toFixed(2);
```

Comment:
This calculates a fraction, not a percent. When you call toFixed, you will end up with a number like "0.23", which means a negligible time will be spent on the first phase. Rather than toFixed, I think you should just multiply by 100.